ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/MessageScrubber.java

```diff
@@ -58,20 +62,14 @@ public static boolean shouldScrub(Map<String, String> attributes, JSONObject jso
         && (attributes.get(Attribute.APP_VERSION).startsWith("68")
             || attributes.get(Attribute.APP_VERSION).startsWith("69"))
         && Optional.of(json) // payload.hangs[].remoteType
-            .map(j -> j.optJSONObject("payload")) //
-            .map(j -> {
-              Boolean isPresent = false;
-              JSONArray arr = j.optJSONArray("hangs");
-              for (int i = 0; i < arr.length(); i++) {
-                String s = arr.optJSONObject(i).optString("remoteType");
-                if (s != null && s.startsWith("webIsolated=")) {
-                  isPresent = true;
-                  break;
-                }
-              }
-              return isPresent;
-            }) //
-            .filter(b -> b).isPresent()) {
+            .map(j -> j.get("payload")) //
+            .map(j -> j.get("hangs")) //
+            .filter(JsonNode::isArray).map(ArrayNode.class::cast) //
+            .map(ArrayNode::elements) //
+            .map(Streams::stream).orElseGet(Stream::empty) //
+            .map(j -> j.get("remoteType")) //
+            .map(JsonNode::textValue) //
+            .anyMatch(s -> s.startsWith("webIsolated="))) {
```

Comment:
calling `JsonNode::textValue` can return null, so this should probably be wrapped in an `Optional`, e.g.
```java
            .anyMatch(j -> Optional.ofNullable(j.path("remoteType").textValue())
                .map(s -> s.startsWith("webIsolated=").isPresent()) //
```

Reply:
`.map(JsonNode::textValue)` will return `Optional.empty` rather than `Optional.of(null)` if the cast that textValue returns null, so it's not necessary to make another explicit `Optional.ofNullable` call.

Reply:
mind=blown

Reply:
but it isn't an optional at that point, it's a stream.

Reply:
Darn. You're totally right. Maybe humans shouldn't be allowed to program.